update-extras.sh

```diff
@@ -83,49 +83,46 @@ fi
 if [[ -f /usr/local/bin/docker-compose ]]; then DOCKER_COMPOSE_V1=true; fi
 if docker compose version &>/dev/null; then DOCKER_COMPOSE_V2=true; fi
 
-# Docker-Compose v1
-if [[ $DOCKER_COMPOSE_V1 == true && $DOCKER_COMPOSE == true ]]; then
-  echo -e "\n*** Updating Docker-Compose v1 (oldstable) ***\n"
-  if COMPOSE=$(find / -name "docker-compose.yaml" >/dev/null 2>&1 | rev | cut -c 20- | rev | tail -n 1); then
-    :
-  elif COMPOSE=$(find / -name "docker-compose.yml" >/dev/null 2>&1 | rev | cut -c 20- | rev | tail -n 1); then
-    :
-  fi
-  cd "$COMPOSE" || exit
-  # Get the containers from the first argument, else get all containers
-  CONTAINER_LIST="${1:-$(docker ps -q)}"
-  for CONTAINER in ${CONTAINER_LIST}; do
-    # Get requirements
-    CONTAINER_IMAGE=$(docker inspect --format "{{.Config.Image}}" --type container "${CONTAINER}")
-    RUNNING_IMAGE=$(docker inspect --format "{{.Image}}" --type container "${CONTAINER}")
-    NAME=$(docker inspect --format "{{.Name}}" --type container "${CONTAINER}" | cut -c 2-)
-    # Pull in the latest version of the container and get the hash
-    docker pull "${CONTAINER_IMAGE}" 2> /dev/null
-    LATEST_IMAGE=$(docker inspect --format "{{.Id}}" --type image "${CONTAINER_IMAGE}")
-    # Restart the container if the image is different by name
-    if [[ ${RUNNING_IMAGE} != "${LATEST_IMAGE}" ]]; then
-      echo "Updating ${CONTAINER} image ${CONTAINER_IMAGE}"
-      /usr/local/bin/docker-compose up -d --no-deps --build "$NAME"
-    fi
-    cd "$COMPOSE" || exit
-    # Docker-Compose v1
-    if [[ $DOCKER_COMPOSE_V1 == true ]]; then
-      echo -e "\n*** Updating Docker-Compose v1 (oldstable) ***\n"
-      /usr/local/bin/docker-compose pull
-      /usr/local/bin/docker-compose up --force-recreate --build -d
-      /usr/local/bin/docker-compose restart
-    fi
-    # Docker-Compose v2
-    if [[ $DOCKER_COMPOSE_V2 == true ]]; then
-      echo -e "\n*** Updating Docker Compose ***"
-      docker compose pull
-      docker compose up -d
+# Docker-Compose new
+if [[ -d "/etc/docker" && $DOCKER_COMPOSE == true ]]; then
+  declare -a COMPOSEFILES=("docker-compose.yaml" "docker-compose.yml" "compose.yaml" "compose.yml")
+  declare -a DIRLIST=()  # Use an array to store directories
+  for COMPOSEFILE in "${COMPOSEFILES[@]}"; do
+    echo "Searching for $COMPOSEFILE..."
+    while IFS= read -r line; do
+      DIRLIST+=("$line")
+    done < <(find /home -name "$COMPOSEFILE" -exec dirname {} \; 2> >(grep -v 'Permission denied'))
```

Comment:
After only testing it, it doesn't found compose file in /root, maybe because it's searching in /home

Reply:
That's what I was thinking, the script could have problem if on a their is a LXC container running portainer trying to update the docker compose contain in portainer